### PR TITLE
JS-2191 USER-2352 Support AWS CDK assertions in S2699

### PR DESCRIPTION
**Pull request by francois-mora-sonarsource**
## Summary

- recognize AWS CDK Template assertions from aws-cdk-lib/assertions in S2699
- support hasResourceProperties and resourceCountIs from Template.fromStack
- cover ordinary, type-aware, and setup-only boundary cases

## Root cause

S2699 did not recognize aws-cdk-lib/assertions as an assertion library, and its Template assertion calls were absent from both assertion-detection paths.

## Validation

- npx tsx --test packages/analysis/src/jsts/rules/S2699/unit.test.ts
- npm run bbf
- npx prettier --check packages/analysis/src/jsts/rules/helpers/assertion-detection.ts packages/analysis/src/jsts/rules/helpers/aws-cdk.ts packages/analysis/src/jsts/rules/S2699/unit.test.ts
- git diff --check a19f17468...HEAD

### CODE REVIEW

File: packages/analysis/src/jsts/rules/helpers/aws-cdk.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ * along with this program; if not, see https://sonarsource.com/license/ssal/
+ */
+import type { Rule } from 'eslint';
+import type estree from 'estree';
+import type { ParserServicesWithTypeInformation } from '@typescript-eslint/utils';
+import ts from 'typescript';
+import { getFullyQualifiedName } from './module.js';
+import { getFullyQualifiedNameTS } from './module-ts.js';
+
+const ASSERTION_METHOD_NAMES = new Set(['hasResourceProperties', 'resourceCountIs']);
```
[the remark this answers is not in the document]
Comment (francois-mora-sonarsource) [reply packages/analysis/src/jsts/rules/helpers/aws-cdk.ts]: Addressed in ed8b16afc. Detection now enumerates documented assertion methods by receiver: Template.fromStack/fromJSON/fromString, Annotations.fromStack, and Tags.fromStack. JavaScript and type-aware tests cover the added APIs, while query/accessor methods remain excluded.

File: packages/analysis/src/jsts/rules/helpers/assertion-detection.ts [hunk 2]
```diff
@@ -157,6 +169,19 @@ export function hasSupportedAssertionLibrary(context: Rule.RuleContext): boolean
...
 
+/**
+ * Like {@link hasSupportedAssertionLibrary}, but also accepts the
+ * {@link UNCLASSIFIED_LIBRARIES}. Only for callers that do not reason about the
+ * script-capable / runner-bound classification.
+ */
+export function hasAssertionLibraryIncludingUnclassified(context: Rule.RuleContext): boolean {
+  return importsOrDependsOnModule(
+    context,
+    [...ASSERTION_LIBRARIES, ...UNCLASSIFIED_LIBRARIES],
```
Comment (guillemsarda) [inline packages/analysis/src/jsts/rules/helpers/assertion-detection.ts:180]: This function is fine for now. However, if a third rule wants to consume a different set of assertion libraries, we should perhaps think of modifying `hasAssertionLibrary` and adding an `allowList` parameter to make it more scalable. 

Right now this scales to one function per different subset of assertion libraries.

**Review by guillemsarda — APPROVED** [review]

File: packages/analysis/src/jsts/rules/helpers/assertion-detection.ts (hunk 2 above)
Comment (francois-mora-sonarsource) [reply packages/analysis/src/jsts/rules/helpers/assertion-detection.ts:180]: I agree, and as discussed, whenever this happens we should start refactoring these shared helpers.
